projects/igniteui-angular/src/lib/banner/banner.component.ts

```diff
@@ -0,0 +1,165 @@
+import { Component, NgModule, EventEmitter, Output, Input, ViewChild, ElementRef, ContentChild } from '@angular/core';
+import { IgxExpansionPanelModule } from '../expansion-panel/expansion-panel.module';
+import { AnimationSettings } from '../expansion-panel/expansion-panel.component';
+import { IgxExpansionPanelComponent } from '../expansion-panel';
+import { IgxIconModule, IgxIconComponent } from '../icon/index';
+import { IToggleView } from '../core/navigation';
+import { IgxButtonModule } from '../directives/button/button.directive';
+import { IgxRippleModule } from '../directives/ripple/ripple.directive';
+import { IgxBannerActionsDirective } from './banner.directives';
+import { CommonModule } from '@angular/common';
+import { CancelableEventArgs } from '../core/utils';
+
+export interface BannerEventArgs {
+    banner: IgxBannerComponent;
+    event?: Event;
+}
+
+export interface BannerCancelEventArgs extends BannerEventArgs, CancelableEventArgs {
+}
+@Component({
+    selector: 'igx-banner',
+    templateUrl: 'banner.component.html'
+})
+export class IgxBannerComponent implements IToggleView {
```

Comment:
@ViktorSlavov A top comment for entire component won't be bad, also small snippets on all the public properties.

Reply:
Thanks for pointing that out! I've added a top comment to the component. All of the public properties already had one (or were @hidden)